test-container/src/main/java/io/strimzi/StrimziZookeeperContainer.java

```diff
@@ -0,0 +1,97 @@
+ * from quay.io/strimzi/kafka. It can be combined with @StrimziKafkaContainer but we suggest to use directly @StrimziKafkaCluster
+ * for more complicated testing.
+ */
+public class StrimziZookeeperContainer extends GenericContainer<StrimziZookeeperContainer> {
+
+    private static final Logger LOGGER = LogManager.getLogger(StrimziZookeeperContainer.class);
+
+    private static final String STARTER_SCRIPT = "/testcontainers_start.sh";
+    private static final List<String> SUPPORTED_KAFKA_VERSIONS = new ArrayList<>(5);
+
+    public static final int ZOOKEEPER_PORT = 2181;
+
+    private int zookeeperExposedPort;
+
+    public StrimziZookeeperContainer(final String imageVersion) {
+        super("quay.io/strimzi/kafka:" + imageVersion);
+        super.withNetwork(Network.SHARED);
+
+        // exposing zookeeper port from the container
+        withExposedPorts(ZOOKEEPER_PORT);
+        withNetworkAliases("zookeeper");
+
+        withEnv("LOG_DIR", "/tmp");
+        withEnv("ZOOKEEPER_CLIENT_PORT", String.valueOf(ZOOKEEPER_PORT));
+        // env for readiness
+        withEnv("ZOO_4LW_COMMANDS_WHITELIST", "rouk");
+    }
+
+    public StrimziZookeeperContainer(Map<String, String> additionalKafkaConfiguration) {
+        this(StrimziKafkaContainer.getStrimziVersion() + "-kafka-" + StrimziKafkaContainer.getLatestKafkaVersion());
+    }
+
+    public StrimziZookeeperContainer() {
+        this(StrimziKafkaContainer.getStrimziVersion() + "-kafka-" + StrimziKafkaContainer.getLatestKafkaVersion());
+    }
+
+    @Override
+    protected void doStart() {
+        // we need it for the startZookeeper(); and startKafka(); to run container before...
+        withCommand("sh", "-c", "while [ ! -f " + STARTER_SCRIPT + " ]; do sleep 0.1; done; " + STARTER_SCRIPT);
+        super.doStart();
+    }
+
+    @Override
+    protected void containerIsStarting(InspectContainerResponse containerInfo, boolean reused) {
+        super.containerIsStarting(containerInfo, reused);
+
+        zookeeperExposedPort = getMappedPort(ZOOKEEPER_PORT);
+
+        LOGGER.info("This is mapped port {}", zookeeperExposedPort);
+
+        String command = "#!/bin/bash \n";
+
+        command += "bin/zookeeper-server-start.sh config/zookeeper.properties\n";
+
+        LOGGER.info("Copying command to 'STARTER_SCRIPT' script.");
+
+        copyFileToContainer(
+            Transferable.of(command.getBytes(StandardCharsets.UTF_8), 700),
+            STARTER_SCRIPT
+        );
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        if (!super.equals(o)) return false;
+        StrimziZookeeperContainer that = (StrimziZookeeperContainer) o;
+        return zookeeperExposedPort == that.zookeeperExposedPort;
+    }
+    @Override
+    public int hashCode() {
+        return Objects.hash(super.hashCode(), zookeeperExposedPort);
```

Comment:
Including `super.hashCode` like this seems to result in a `hashCode()` that's incompatible with `equals()`.